recipe/meta.yaml

```
@@ -30,13 +30,20 @@ source:
   sha256: 908742d8f3c6fdd6d1d6316a7b919b5c506474f9551f491aa7335cb4f50bffbd  # [win]
 
 build:
-  number: 1
+  number: 2
   skip: true  # [osx]
 
 test:
   commands:
     - test -d $PREFIX/lib/cmake                      # [linux]
     - test -d $PREFIX/targets/{{ target_name }}/lib  # [linux]
+    - if not exist %LIBRARY_LIB%\{{ target_name }}\cmake exit 1    # [win]
+    - if not exist %LIBRARY_INC%\targets\{{ target_name }} exit 1  # [win]
+    # Extracts the correct cccl version from Nvidia's cuda_cccl archive, and compares it against the version of the cccl conda package
+    - CCCL_VERSION_MAJOR=$(grep "set(CCCL_VERSION_MAJOR" $PREFIX/targets/{{ target_name }}/lib/cmake/cccl/cccl-config-version.cmake | cut -d " " -f2 | tr -d ')')   # [linux]
+    - CCCL_VERSION_MINOR=$(grep "set(CCCL_VERSION_MINOR" $PREFIX/targets/{{ target_name }}/lib/cmake/cccl/cccl-config-version.cmake | cut -d " " -f2 | tr -d ')')   # [linux]
+    - CCCL_VERSION_PATCH=$(grep "set(CCCL_VERSION_PATCH" $PREFIX/targets/{{ target_name }}/lib/cmake/cccl/cccl-config-version.cmake | cut -d " " -f2 | tr -d ')')   # [linux]
+    - if [[ "${CCCL_VERSION_MAJOR}.${CCCL_VERSION_MINOR}.${CCCL_VERSION_PATCH}" != {{ cccl_version }} ]]; then exit 1; fi                                           # [linux]
```

Do you think this would be clearer? (I tested the individual commands but not in a full conda-build)
```suggestion
    - CCCL_VERSION_FILE="$PREFIX/targets/{{ target_name }}/lib/cmake/cccl/cccl-config-version.cmake"  # [linux]
    - CCCL_VERSION_MAJOR=$(sed -n 's/set(CCCL_VERSION_MAJOR \([0-9]*\))/\1/p')  # [linux]
    - CCCL_VERSION_MINOR=$(sed -n 's/set(CCCL_VERSION_MINOR \([0-9]*\))/\1/p')  # [linux]
    - CCCL_VERSION_PATCH=$(sed -n 's/set(CCCL_VERSION_PATCH \([0-9]*\))/\1/p')  # [linux]
    - [[ "${CCCL_VERSION_MAJOR}.${CCCL_VERSION_MINOR}.${CCCL_VERSION_PATCH}" == "{{ cccl_version }}" ]] || exit 1  # [linux]
```

---

yes thanks, this seems safer